aten/src/ATen/native/Normalization.cpp

```diff
@@ -365,9 +365,13 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_backward_cpu_template(
   for (const auto i : c10::irange(2, ndim)) {
     reduce_dims[i - 1] = i;
   }
-
-  auto sum = at::sum(grad_out_, /*dim=*/reduce_dims);
-  auto sum_a = sum.accessor<scalar_t, 1>();
+  // Using float data type for Half sum to avoid overflow
+  // since the representation range of Half is small.
+  auto sum = grad_out_.scalar_type() == kHalf
```

Comment:
I think an if/else here might be more readable than a ternary expression

Reply:
Can I keep the ternary expression? If use if else, the code will become longer like：
```
Tensor sum;
if (grad_out_.scalar_type() == kHalf) {
  sum = at::sum(grad_out_.to(ScalarType::Float), /*dim=*/reduce_dims);
} else {
  sum = at::sum(grad_out_, /*dim=*/reduce_dims);
}
```

Reply:
Sure, either one should work. I personally don't mind more verbose / longer code if it's more readable and easier to maintain.